src/sinks/gcp_chronicle/compression.rs

```diff
@@ -0,0 +1,153 @@
+use serde::{de, ser};
+use serde_json::Value;
+use std::{cell::RefCell, collections::BTreeSet};
+use vector_lib::configurable::ToValue;
+
+use indexmap::IndexMap;
+use vector_lib::configurable::attributes::CustomAttribute;
+use vector_lib::configurable::{
+    schema::{
+        apply_base_metadata, generate_const_string_schema, generate_one_of_schema,
+        generate_struct_schema, get_or_generate_schema, SchemaGenerator, SchemaObject,
+    },
+    Configurable, GenerateError, Metadata,
+};
+
+use crate::sinks::util::buffer::compression::CompressionLevel;
+use crate::sinks::util::Compression;
+
+/// Compression configuration.
+#[derive(Clone, Copy, Debug, Derivative, Eq, PartialEq)]
+#[derivative(Default)]
+pub enum ChronicleCompression {
+    /// No compression.
+    #[derivative(Default)]
+    None,
+
+    /// [Gzip][gzip] compression.
+    ///
+    /// [gzip]: https://www.gzip.org/
+    Gzip(CompressionLevel),
+}
+
+impl From<ChronicleCompression> for Compression {
+    fn from(compression: ChronicleCompression) -> Self {
+        match compression {
+            ChronicleCompression::None => Compression::None,
+            ChronicleCompression::Gzip(compression_level) => Compression::Gzip(compression_level),
+        }
+    }
+}
+
+impl TryFrom<Compression> for ChronicleCompression {
+    type Error = String;
+
+    fn try_from(compression: Compression) -> Result<Self, Self::Error> {
+        match compression {
+            Compression::None => Ok(ChronicleCompression::None),
+            Compression::Gzip(compression_level) => {
+                Ok(ChronicleCompression::Gzip(compression_level))
+            }
+            _ => Err("Compression type is not supported by Chronicle".to_string()),
+        }
+    }
+}
+
+// Schema generation largely copied from `src/sinks/util/buffer/compression`
+impl Configurable for ChronicleCompression {
+    fn metadata() -> Metadata {
+        let mut metadata = Metadata::default();
+        metadata.set_title("Compression configuration.");
+        metadata.set_description("All compression algorithms use the default compression level unless otherwise specified.");
+        metadata.add_custom_attribute(CustomAttribute::kv("docs::enum_tagging", "external"));
+        metadata.add_custom_attribute(CustomAttribute::flag("docs::advanced"));
+        metadata
+    }
+
+    fn generate_schema(gen: &RefCell<SchemaGenerator>) -> Result<SchemaObject, GenerateError> {
```

Comment:
Could any of this be refactored out into functions shared with `Compression`? This looks like a lot of duplication just with different lists.

@pront are there other functions that duplicate parts of this as well? Maybe that kind of clean up is out of scope here.

Reply:
I agree there's potential for code sharing here. That being said, I am happy to accept this PR as is. We can add a comment here as a reminder to follow up and refactor.